src/engine.rs

```diff
@@ -288,6 +344,58 @@ impl Engine {
         sdk::sha256(&data)
     }
 
+    pub fn charge_gas_limit(
+        sender: &Address,
+        gas_limit: U256,
+        gas_price: U256,
+    ) -> Result<Wei, GasPaymentError> {
+        // Early exit as performance optimization
+        if gas_price.is_zero() {
+            return Ok(Wei::zero());
+        }
+
+        let eth_amount = Wei::new(
```

Comment:
This is not `eth_amount`, this is indeed `wei_amount` and payment for gas. Please let's rename to `payment_for_gas`:

```suggestion
        let payment_for_gas = Wei::new(
```

Reply:
Done in https://github.com/aurora-is-near/aurora-engine/pull/215/commits/5bdc6d0dc4897e5892f0f72d8e1eee36602a45e0